### PR TITLE
Add invite user

### LINKED ISSUE #3191: Create flow for user creation via email invite
This is for the basic functionality, A proper design issue will follow

### PR DESCRIPTION
**Pull request by shimi9276**
Adding the Backend functionality for lakefs admins to invite a user.
Closes #3191

### CODE REVIEW

File: api/swagger.yml [hunk 1]
```diff
@@ -556,6 +556,17 @@ components:
         email:
           type: string
 
+    InviteUserRequest:
+      type: object
+      required:
+        - email
+        - id
```
Comment (talSofer) [inline api/swagger.yml:563]: Why does an invite user request require an id? how would the user have an id at this point?

Comment (shimi9276) [reply api/swagger.yml:563]: The id here is just the name of the user (Just like when creating a user, see auth/user/). The idea here is to preserve backend compatibility for createUser as much as possible.

**Review by nopcoder — CHANGES_REQUESTED** [review]
discussed the flow of invite that will match password reset - waiting for the next pr

**Comment by shimi9276** [comment]
Closed this one and opened a new and updated one 
See [here](https://github.com/treeverse/lakeFS/pull/3220#)

**Pull request closed without merging**